modules/diskq/qdisk.c

```diff
@@ -77,6 +77,7 @@ struct _QDisk
   gint64 file_size;
   QDiskFileHeader *hdr;
   DiskQueueOptions *options;
+  GString *temp;
```

Comment:
Adding such a state to QDisk is not a good idea in my opinion.

If we needed a string buffer as a class field, we would have to find a more eloquent name for that, but I think we can do better by using scratch buffers instead. They were made exactly for this use case.

For example:
https://github.com/syslog-ng/syslog-ng/blob/e0104231703d5d68091183f24696e83ba014a722/modules/xml/xml.c#L61-L87

The main advantage in this context would be keeping string buffers local on hot paths.
Disk-buffer is a multi-threaded entity, its methods are called from multiple source threads and a destination thread as well. We currently have top-level mutexes that prevent race conditions inside QDisk, but that can eventually be changed. 

Moving a local string buffer into the `QDisk` struct complicates things further when considering the parallel access of this field.